app/liquid_tags/glitch_tag.rb

```diff
@@ -31,49 +37,70 @@ def render(_context)
 
   private
 
-  def valid_id?(input)
-    (input =~ ID_REGEXP)&.zero?
+  def parsed_input(input)
+    id, *options = input.split
+    stripped_id = id.gsub(/^~/, "") # remove possible preceeding tilde
+    match = pattern_match_for(stripped_id, REGEXP_OPTIONS)
+    raise StandardError, I18n.t("liquid_tags.glitch_tag.invalid_glitch_id") unless match
+
+    [get_slug(match), parse_options(options, match)]
   end
 
-  def parse_id(input)
-    id = input.split.first
-    id.sub!(TILDE_PREFIX_REGEXP, "")
-    raise StandardError, I18n.t("liquid_tags.glitch_tag.invalid_glitch_id") unless valid_id?(id)
+  def get_slug(match)
+    if match_has_named_capture_group?(match, "subdomain")
+      match[:subdomain].presence
+    else
+      match[:slug]
+    end
+  end
 
-    id
+  def match_has_named_capture_group?(match, group_name)
```

Comment:
NOW I remember why I had this method!

When the liquid is generated with the `glitch` keyword, the slug will match the `ID_REGEXP`, which lacks a `subdomain` capture group:
```ruby
#<MatchData "zircon-quixotic-attraction" slug:"zircon-quixotic-attraction">
```
The `get_slug` method (which handles matches from both `ID_REGEXP` and `REGISTRY_REGEXP`) thus needs to ascertain the presence of this capture group, else the specs in `spec/liquid_tags/glitch_tag_spec.rb` all fail with:
```
IndexError:
        undefined group name reference: <name of missing capture group>
```

I'm all ears for cleverer solutions. 🙏🏾